lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -1,10 +1,36 @@
 class ActiveModel::Serializer::Adapter::JsonApi < ActiveModel::Serializer::Adapter
+        # JsonApiObject: namespace for functions that build
+        # JSON API objects
+        module JsonApiObject
+          # Make JSON API top-level jsonapi member opt-in
+          # ref: http://jsonapi.org/format/#document-top-level
+          ActiveModel::Serializer.config.jsonapi_toplevel_member = false
+          ActiveModel::Serializer.config.jsonapi_version = '1.0'
+          ActiveModel::Serializer.config.jsonapi_toplevel_meta = {}
+          module JsonApi
+            module_function
+
+            def add!(document)
+              return document unless ActiveModel::Serializer.config.jsonapi_toplevel_member
+              object = {
+                jsonapi: {
+                  version: ActiveModel::Serializer.config.jsonapi_version,
+                  meta: ActiveModel::Serializer.config.jsonapi_toplevel_meta
+                }
+              }
+              object[:jsonapi].reject! { |_, v| v.blank? }
+              document.merge!(object)
+            end
+          end
+        end
         extend ActiveSupport::Autoload
         autoload :PaginationLinks
         autoload :FragmentCache
 
         def initialize(serializer, options = {})
           super
+          @hash = {}
+          JsonApiObject::JsonApi.add!(hash)
```

Comment:
I'd much rather have it as a conditional assignation than a magical `add!` call that may or may not add stuff to its argument. Moreover, I think it belongs to the `serializable_hash` method rather than the constructor. How about:
```ruby
hash[:jsonapi] = JsonApiObject.serializable_hash if ActiveModel::Serializer.config.jsonapi_toplevel_member
```

Plus I'm starting to wonder whether the config should be named `jsonapi_toplevel_member` or something more explicit like `include_jsonapi_object`.